posthog-react-native/src/storage.ts

```diff
@@ -1,22 +1,34 @@
-import { PostHogCustomAsyncStorage } from './types'
+import { PostHogStorage } from './types'
 
 const POSTHOG_STORAGE_KEY = '.posthog-rn.json'
 const POSTHOG_STORAGE_VERSION = 'v1'
 
 type PostHogStorageContents = { [key: string]: any }
 
-// NOTE: The core prefers a synchronous storage so we mimic this by pre-loading all keys
-export class SemiAsyncStorage {
+export abstract class SemiAsyncStorage {
```

Comment:
This is what a custom "semi async storage" should implement, and what the now called PostHogSemiAsyncStorage implements